src/main/java/at/ac/uibk/dps/cirrina/runtime/job/JobMonitor.java

```diff
@@ -112,7 +112,15 @@ private void nodeCreated(String nodePath, byte[] data) {
         // Call listener
         jobListener.newJob(job);
       } catch (UnsupportedOperationException | IllegalArgumentException e) {
-        logger.error("Failed to register job", e);
+        logger.error("Failed to parse job description for job '{}'. Skipping this job.", jobName, e);
+
+        try {
+          // Delete the invalid job
+          curatorFramework.delete().forPath(nodePath);
```

Comment:
Here, _job.delete_ should be called.

Reply:
used new Job(jobName, nodePath, null, curatorFramework).delete();
as the jobDescription is invalid we construct a new job with `null` for the jobDescription